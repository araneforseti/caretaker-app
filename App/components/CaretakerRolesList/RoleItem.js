import React from 'react';
import PropTypes from 'prop-types';
import {
  Switch,
  TouchableOpacity,
} from 'react-native';
import renderif from '../../utils/renderif';
import theme from '../../config/theme';

import container from './RoleItem.container';
import ListItem from '../material-ui/ListItem';

export class RoleItem extends React.PureComponent {
  render() {
    const {
      name,
      description,
      hasRole,
      isHelper,
      onLongPress,
      onToggleSwitch,
      isLast
    } = this.props;

    return <TouchableOpacity onLongPress={onLongPress}>
      <ListItem
        divider={isLast}
        centerElement={ {
          primaryText: name,
          secondaryText: description
        } }
        rightElement={renderif(isHelper, <Switch
          value={hasRole}
          onValueChange={onToggleSwitch}
          onTintColor={theme.palette.lightColor}
          thumbTintColor={hasRole ? theme.palette.secondaryColor : null}
        />)}
      />
    </TouchableOpacity>;
  }
}

RoleItem.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  hasRole: PropTypes.bool,
  isHelper: PropTypes.bool,
  isLast: PropTypes.bool,

//From Container
  onLongPress: PropTypes.func,
  onToggleSwitch: PropTypes.func
};

export default container(RoleItem);
