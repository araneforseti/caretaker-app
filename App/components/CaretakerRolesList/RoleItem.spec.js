import React from 'react';
import renderer from 'react-test-renderer';
import RoleItem from './RoleItem';

describe('<RoleItem />', () => {
  let props;
  const roleItem = () => renderer.create(<RoleItem {...props} />);

  beforeEach(() => {
    props = {
      name: 'driver',
      description: 'Drives around and stuff',
      hasRole: true,
      isHelper: false,
      isLast: true,
    };
  });

  it('should render correctly', () => {
    expect(roleItem()).toMatchSnapshot();
  });

});
