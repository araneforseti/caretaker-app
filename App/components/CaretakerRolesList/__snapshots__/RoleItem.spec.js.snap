// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoleItem /> should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <ThemeProvider
    uiTheme={
      Object {
        "palette": Object {
          "accentColor": "#FF7575",
          "alternateTextColor": "white",
          "lightColor": "#FFC0C0",
          "primaryColor": "#FF7575",
          "primaryTextColor": "black",
          "secondaryColor": "#CC5D5D",
        },
        "spacing": Object {
          "miniActionButtonSize": 42,
        },
        "toolbar": Object {
          "container": Object {
            "height": 50,
          },
        },
        "typography": Object {
          "fontSize": Object {
            "display": 25,
          },
          "subheading": Object {
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 24,
          },
        },
      }
    }
  />
</View>
`;
